x/ecocredit/core/genesis_test.go

```diff
@@ -0,0 +1,414 @@
+		},
+		{
+			"valid test case",
+			func() json.RawMessage {
+				return json.RawMessage(`{
+				"regen.ecocredit.v1.ClassInfo":[{
+					"name":"1",
+					"admin":"OfVGZ+vChK/1gQfbXZ6rxsz3QNQ=",
+					"credit_type":"C"
+				}],
+				"regen.ecocredit.v1.ProjectInfo":[{
+					"name":"01",
+					"admin":"OfVGZ+vChK/1gQfbXZ6rxsz3QNQ=",
+					"class_id":"1",
+				}],
+				"regen.ecocredit.v1.BatchInfo":[{
+					"issuer":"OfVGZ+vChK/1gQfbXZ6rxsz3QNQ=",
+					"project_id":"1",
+					"batch_denom":"1/2",
+					"metadata":"meta-data"
+				}],
+				"regen.ecocredit.v1.BatchBalance":[
+					{
+						"address":"Ak5WDUYGfdv4gNMF500MFF86NWA=",
+						"batch_id":"1",
+						"tradable":"100.123",
+						"retired":"100.123"
+					},
+					{
+						"address":"OfVGZ+vChK/1gQfbXZ6rxsz3QNQ=",
+						"batch_id":"1",
+						"tradable":"100.123",
+						"retired":"100.123",
+					}
+				],
+				"regen.ecocredit.v1.BatchSupply":[
+					{
+						"batch_id":"1",
+						"tradable_amount":"200.246",
+						"retired_amount":"200.246"
+					}
+				]
+			}`)
+			},
+			defaultParams,
+			false,
+			"",
+		},
+		{
+			"valid test case, multiple classes",
+			func() json.RawMessage {
+				return json.RawMessage(`
+				{
+					"regen.ecocredit.v1.ClassInfo": [
+					  {
+						"name": "1",
+						"admin": "OfVGZ+vChK/1gQfbXZ6rxsz3QNQ=",
+						"credit_type": "C"
+					  },
+					  {
+						"name": "2",
+						"admin": "Ak5WDUYGfdv4gNMF500MFF86NWA=",
+						"credit_type": "C"
+					  }
+					],
+					"regen.ecocredit.v1.ProjectInfo": [
+					  {
+						"name": "01",
+						"admin": "OfVGZ+vChK/1gQfbXZ6rxsz3QNQ=",
+						"class_id": "1"
+					  },
+					  {
+						"name": "03",
+						"admin": "Ak5WDUYGfdv4gNMF500MFF86NWA=",
+						"class_id": "2"
+					  }
+					],
+					"regen.ecocredit.v1.BatchInfo": [
+					  {
+						"issuer": null,
+						"project_id": "1",
+						"batch_denom": "1/2"
+					  },
+					  {
+						"issuer": null,
+						"project_id": "2",
+						"batch_denom": "2/2"
+					  }
+					],
+					"regen.ecocredit.v1.BatchBalance": [
+					  {
+						"address": "Ak5WDUYGfdv4gNMF500MFF86NWA=",
+						"batch_id": "1",
+						"tradable": "100.123",
+						"retired": "100.123"
+					  },
+					  {
+						"address": "OfVGZ+vChK/1gQfbXZ6rxsz3QNQ=",
+						"batch_id": "2",
+						"tradable": "100.123",
+						"retired": "100.123"
+					  }
+					],
+					"regen.ecocredit.v1.BatchSupply": [
+					  {
+						"batch_id": "1",
+						"tradable_amount": "100.123",
+						"retired_amount": "100.123"
+					  },
+					  {
+						"batch_id": "2",
+						"tradable_amount": "100.123",
+						"retired_amount": "100.123"
+					  }
+					]
+				  }
+				`)
+			},
+			defaultParams,
+			false,
+			"",
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+			err := core.ValidateGenesis(tc.gensisState(), tc.params)
+			if tc.expectErr {
+				require.Error(t, err)
+				require.Contains(t, err.Error(), tc.errorMsg)
```

Comment:
Looks like you are using the full error for all test cases. You should be able to use `EqualError` here:

```suggestion
				require.EqualError(t, err, tc.errorMsg)
```

Reply:
Actual error is even longer `invalid creator address: decoding bech32 failed: invalid separator index -1: invalid address ...`